src/rpc/blockchain.cpp

```diff
@@ -399,7 +413,13 @@ static std::string EntryDescriptionString()
            "    \"spentby\" : [           (array) unconfirmed transactions spending outputs from this transaction\n"
            "        \"transactionid\",    (string) child transaction id\n"
            "       ... ]\n"
-           "    \"bip125-replaceable\" : true|false,  (boolean) Whether this transaction could be replaced due to BIP125 (replace-by-fee)\n";
+           "    \"bip125-replaceable\" : true|false,  (boolean) Whether this transaction could be replaced due to BIP125 (replace-by-fee)\n"
+           "    \"replaceable\" :         (bool) Whether this transaction is replaceable.\n"
+           "    \"replaceable-reason\" :  (string) Why this transaction is replaceable.\n"
+           "                              If the transaction is not replaceable, this is Null.\n"
```

Comment:
Have you considered not adding the key at all?

Reply:
The caller doesn't have to use it. I think it might be helpful to see the reason why something is replaceable, no?

Reply:
Ops, I mean when it's null.

Reply:
A missing key will throw `KeyError` for the caller

Reply:
That depends on the caller's programming language and API of choice.

If they choose a language that gives `KeyError`, then they should expect it...

Reply:
I think it is not a good api design to make parsing depend on the result

Reply:
No strong feeling but it was already suggested (and implemented) elsewhere to omit keys when not applicable. For instance, search for "present" in rpcwallet.cpp.

Reply:
Lets discuss this in #17155

Reply:
Closing this discussion